app/javascript/packs/application.js

```diff
@@ -3,4 +3,23 @@ require("turbolinks").start()
 require("@rails/activestorage").start()
 require("channels")
 require("bootstrap/dist/js/bootstrap")
+
+$(function () {
+    $document.getElementById('search-text').keyup(function(){
```

Comment:
Chromeの検証ツールを開いて，エラーを確認していくようにしましょう。書き方がいろいろ間違っていますよ。
`JavaScript` で書くならばこうなります。

```suggestion
    document.getElementById('search-text').addEventListener('keyup', () => {
```